### DISCUSSION TITLE
Provide easy way to test user-defined message handlers

### DISCUSSION THREAD

**Discussion by stijnmoreels** [arcus-azure/arcus.messaging#155] [Ideas]
After the routing is determined #141 , we could maybe rethink our integration tests as they are all relying on external dependencies (Azure Service Bus & Event Grid) to test sometimes not dependency-related stuff but more message handling functionality.

We could change some of our tests to only test the necessary functionality by sending our own test messages directly in the router instead of going via the Azure Service Bus. Of course we can still have some tests that do go through Azure Service Bus and the message pump, but if we extract also some of these message handling tests, we can more extensively test by sending more 'extreme' kind of messages with maybe malicious inputs which we cannot do right now.

**Comment by tomkerkhove** [comment]
Let's move this to an issue and plan for it
